### PR TITLE
feat: upgade Profile and Contacts to new Slashtags

### LINKED ISSUE #706: [Feature]: Perform full backup
### Describe the problem

Bitkit currently does not save all of the user's data to the remote backup.

### Describe the solution

Should save/restore all (useful) user data:

- [x] Lightning backup
- [ ] Profile
- [ ] Contacts
- [x] Widgets
- [x] Tags
- [x] Preferences

### PR DESCRIPTION
**Pull request by limpbrains**
### Description

Upgrade Profile and Contacts
Move webrelay staff to SlashtagsProvider2.

### Linked Issues/Tasks

closes #706
https://app.asana.com/0/1205132561089038/1205132561089038

### Type of change

Refactoring

### Tests

Detox test

### CODE REVIEW

File: src/navigation/onboarding/OnboardingNavigator.tsx [hunk 1]
```diff
@@ -32,6 +35,12 @@ const navOptionHandler = {
 };
 
 const OnboardingNavigator = (): ReactElement => {
+	const selectedNetwork = useSelector(selectedNetworkSelector);
+
+	useEffect(() => {
+		connectToElectrum({ selectedNetwork });
+	}, [selectedNetwork]);
+
```
Comment (pwltr) [inline src/navigation/onboarding/OnboardingNavigator.tsx:43]: What's the reason to connect to electrum in the onboarding?

File: src/screens/Contacts/Contact.tsx [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { FadeIn, FadeOut } from 'react-native-reanimated';
 import Clipboard from '@react-native-clipboard/clipboard';
 import Share from 'react-native-share';
 import { useTranslation } from 'react-i18next';
+import { SlashURL } from '@synonymdev/slashtags-sdk';
```
Comment (pwltr) [inline src/screens/Contacts/Contact.tsx]: Should this use the new `@synonymdev/slashtags-url` instead?

File: src/screens/Contacts/Contact.tsx [hunk 3]
```diff
@@ -72,9 +74,9 @@ const Contact = ({
 	};
 
 	const onDelete = useCallback(() => {
-		deleteContact(slashtag, url);
+		deleteContact(url);
 		navigation.navigate('Contacts');
-	}, [navigation, slashtag, url]);
+	}, [navigation, url]);
```
Comment (pwltr) [inline src/screens/Contacts/Contact.tsx:79]: Can probably enable the delete button again with this PR? It's commented out further down

File: src/store/reducers/slashtags.ts [hunk 4]
```diff
@@ -66,6 +66,43 @@ const slashtags = (
...
+		case actions.CONTACT_DELETE: {
+			const contacts = state.contacts;
+			delete contacts[action.payload.id];
+			return {
+				...state,
+				contacts: {
+					...contacts,
+				},
+			};
+		}
```
Comment (pwltr) [inline src/store/reducers/slashtags.ts]: This is mutating state, either copy `contacts` first or use `removeKeysFromObject` helper

File: src/utils/backup/backups-subscriber.tsx [hunk 5]
```diff
@@ -172,6 +172,36 @@ const EnabledSlashtag = (): ReactElement => {
...
+
+			performRemoteBackup({
+				slashtag,
+				isSyncedKey: 'remoteSlashtagsBackupSynced',
+				syncRequiredKey: 'remoteSlashtagsBackupSyncRequired',
+				syncCompletedKey: 'remoteSlashtagsBackupLastSync',
+				backupCategory: EBackupCategories.slashtags,
+				selectedNetwork,
+				backup: back,
+			}).then();
```
Comment (pwltr) [inline src/utils/backup/backups-subscriber.tsx:194]: Can we close [this issue](https://github.com/synonymdev/bitkit/issues/706) with this change?

**Review by pwltr — CHANGES_REQUESTED** [review]

File: src/navigation/onboarding/OnboardingNavigator.tsx (hunk 1 above)
Comment (limpbrains) [reply src/navigation/onboarding/OnboardingNavigator.tsx:43]: Wallet will be created faster, especially during e2e tests

File: src/screens/Contacts/Contact.tsx (hunk 2 above)
Comment (limpbrains) [reply src/screens/Contacts/Contact.tsx]: fixed

File: src/screens/Contacts/Contact.tsx (hunk 3 above)
Comment (limpbrains) [reply src/screens/Contacts/Contact.tsx:79]: next, PR, I need to test it proparly

File: src/utils/backup/backups-subscriber.tsx (hunk 5 above)
Comment (limpbrains) [reply src/utils/backup/backups-subscriber.tsx:194]: I think we can

File: src/store/reducers/slashtags.ts (hunk 4 above)
Comment (limpbrains) [reply src/store/reducers/slashtags.ts]: I can't seed what is wrong, but used removeKeysFromObject anyway

**Review by pwltr — APPROVED** [review]

**Pull request merged**